provider/internal/image.go

```diff
@@ -885,17 +897,17 @@ func (i *Image) Delete(
 		multierr = errors.Join(multierr, err)
 	}
 
-	return multierr
+	return infer.DeleteResponse{}, multierr
 }
 
 // Diff re-implements most of the default diff behavior, with the exception of
 // ignoring "password" changes on registry inputs.
 func (*Image) Diff(
 	_ context.Context,
-	_ string,
-	olds ImageState,
-	news ImageArgs,
+	req infer.DiffRequest[ImageArgs, ImageState],
 ) (provider.DiffResponse, error) {
+	olds, news := req.Olds, req.News
```

Comment:
olds/news feels very "inside baseball." IMO it would be reasonable to call these State/Inputs for consistency.